clients/cli/src/sync_metadata_to_token2022.rs

```diff
@@ -0,0 +1,156 @@
+use {
+    crate::{
+        common::{parse_pubkey, process_transaction},
+        config::Config,
+        output::{format_output, println_display},
+        CommandResult,
+    },
+    clap::{ArgMatches, Args},
+    mpl_token_metadata::accounts::Metadata as MetaplexMetadata,
+    serde_derive::{Deserialize, Serialize},
+    serde_with::{serde_as, DisplayFromStr},
+    solana_cli_output::{display::writeln_name_value, QuietDisplay, VerboseDisplay},
+    solana_pubkey::Pubkey,
+    solana_remote_wallet::remote_wallet::RemoteWalletManager,
+    solana_signature::Signature,
+    solana_signer::Signer,
+    solana_transaction::Transaction,
+    spl_token_wrap::{
+        get_wrapped_mint_address, get_wrapped_mint_authority,
+        instruction::sync_metadata_to_token_2022,
+    },
+    std::{
+        fmt::{Display, Formatter},
+        rc::Rc,
+    },
+};
+
+#[derive(Clone, Debug, Args)]
+pub struct SyncMetadataToToken2022Args {
+    /// The address of the unwrapped mint whose metadata will be synced from
+    #[clap(value_parser = parse_pubkey)]
+    pub unwrapped_mint: Pubkey,
+
+    /// Specify that the source metadata is from a `Metaplex` Token Metadata
+    /// account. The CLI will derive the PDA automatically.
+    #[clap(long)]
+    pub metaplex: bool,
+
+    /// Optional source metadata account when the unwrapped mint's metadata
+    /// pointer points to an external account or third-party program
+    #[clap(long, value_parser = parse_pubkey, conflicts_with = "metaplex", requires = "program-id")]
+    pub metadata_account: Option<Pubkey>,
+
+    /// Optional owner program for the source metadata account, when owned by a
+    /// third-party program
+    #[clap(long, value_parser = parse_pubkey)]
+    pub program_id: Option<Pubkey>,
```

Comment:
nit: Typically the `--program-id` arg is to specify the address of the program to target, ie the token-wrap program in this case.

I'd go with something like `--metadata-program-id` so that we can eventually specify `--program-id` globally in the CLI